### ISSUE TITLE
Add raw pointer interface

### ISSUE THREAD

**Issue by remexre** [Manishearth/rust-gc#62]
Would there be significant downsides/dangers to having methods:

```rust
fn into_raw(gc: Gc<T>) -> *const T { ... }
unsafe fn from_raw(ptr: *const T) -> Gc<T> { ... }
```

analogously to [`Rc::into_raw`](https://doc.rust-lang.org/std/rc/struct.Rc.html#method.into_raw), [`Rc::from_raw`](https://doc.rust-lang.org/std/rc/struct.Rc.html#method.from_raw)?

I want to store a `Gc`'d value across an FFI boundary (for use in a callback), while keeping it rooted; are there any huge dangers here versus doing the same with an `Rc`?

The alternative would be boxing up the `Gc`, then going `Box<Gc<T>> -> *const Gc<T>` and converting it back to a `Box<Gc<T>>` once I'm on the Rust side, I suppose. Either way, it'd be nice to have the `{into,from}_raw` functions.

**Comment by ravernkoh** [comment]
Seconded.

I am currently implementing a toy language in Rust and would love the ability to identify each GC-ed object.

My makeshift solution is to implement some form of global counter to assign IDs, which feels very hacky. I also have to keep track when objects are freed so as to re-use these ids.

**Comment by mystor** [comment]
Due to the way rooting is handled in `rust-gc` currently, this should be possible. It's unlikely we're going to change the API for rooting anytime soon, even if there may be other better systems.

I think I'd merge a PR adding `Gc::{into,from}_raw` :-)

**Comment by ravernkoh** [comment]
Great!

I’m not very experienced in Rust but I’ll try my best :)

**Comment by ravernkoh** [comment]
I think we can close this issue already :)

**Comment by andersk** [comment]
This was fixed in #71.
